app/assets/javascripts/darkswarm/services/dereferencer.js.coffee

```diff
@@ -1,6 +1,8 @@
 Darkswarm.factory 'Dereferencer', ->
   new class Dereferencer
-    dereference: (array, data)->
+    dereference: (array, data) ->
       if array
         for object, i in array
-          array[i] = data[object.id]
```

Comment:
I rewrote this in a later commit with if-then-else. Coffeescript doesn't have the `?` operator while javascript has. http://stackoverflow.com/questions/10146080/ternary-operation-in-coffeescript